### Data Loading

IDD 20k dataset

In [ ]:
from collections import Counter
import glob
import json
from pathlib import Path

all_labels = Counter()
json_files = glob.glob("../data/raw/idd20kII/gtFine/**/**/*.json", recursive=True)
print(len(json_files))
for jf in json_files:  # sample first
    with open(jf) as f:
        d = json.load(f)
    for obj in d.get("objects", []):
        all_labels[obj["label"]] += 1

print(all_labels.most_common(50))

24267
[('obs-str-bar-fallback', 517908), ('vegetation', 468180), ('pole', 314661), ('billboard', 149130), ('motorcycle', 141576), ('rider', 123678), ('person', 102900), ('car', 89970), ('building', 84759), ('vehicle fallback', 64992), ('autorickshaw', 47211), ('drivable fallback', 46461), ('wall', 34107), ('truck', 33429), ('non-drivable fallback', 33192), ('curb', 31974), ('traffic sign', 29766), ('sky', 25191), ('road', 25164), ('fence', 12123), ('animal', 11889), ('bus', 10149), ('guard rail', 7782), ('sidewalk', 7272), ('fallback background', 7191), ('bicycle', 3222), ('bridge', 3165), ('polegroup', 2961), ('traffic light', 2340), ('out of roi', 756), ('caravan', 360), ('parking', 90), ('rail track', 63), ('rectification border', 45), ('train', 39), ('trailer', 30), ('tunnel', 15), ('ego vehicle', 12), ('ground', 9), ('unlabeled', 6), ('license plate', 6)]


In [34]:
all_labels["license plate"]

6

out of 24267 images only 6 have labelled license plate. Not useful

In [8]:
import ndjson
from pathlib import Path
path = Path(r"..\data\raw\plate.ndjson")
with open(path) as f:
    data = ndjson.load(f)
data

[{'type': 'dataset',
  'task': 'detect',
  'name': 'plate',
  'description': '',
  'bytes': 733364994,
  'url': 'https://platform.ultralytics.com/ultralytics-test/datasets/plate',
  'class_names': {'0': 'license_plate'},
  'version': 'latest',
  'created_at': '2026-07-07T09:03:50.709Z',
  'updated_at': '2026-07-07T09:04:56.120Z'},
 {'type': 'image',
  'file': 'Screenshot-2025-03-30-at-12_43_00-AM_png.rf.f2cc6cc995198026e0b304cf04b7a9e5.jpg',
  'url': 'https://cdn.ul.run/eu/i/e6a743d7f39074b2cdb06801c69afc4a.jpg?Expires=1784020001&KeyName=key-v1&Signature=EZ6IxLhSeJ7We7UxyCb7j1Ir7ZQ',
  'width': 640,
  'height': 640,
  'split': 'train'},
 {'type': 'image',
  'file': 'Screenshot-2025-03-09-at-11_27_19-PM_png.rf.4226f4c0e523f6160d0eb17544705ea3.jpg',
  'url': 'https://cdn.ul.run/eu/i/d4fc56344f25834ffc3b79349b9b2217.jpg?Expires=1784020001&KeyName=key-v1&Signature=W0FAIZ2xt-Q66Omr5W42AybtK-8',
  'width': 640,
  'height': 640,
  'split': 'train',
  'annotations': {'boxes': [[0, 0.10716, 0.3

In [6]:
path.exists()

False

In [1]:
import ndjson

In [2]:
!python -m ipykernel install --user --name=venv

Installed kernelspec venv in C:\Users\sys_a\AppData\Roaming\jupyter\kernels\venv


In [9]:
from ultralytics import YOLO

# Load a model
model = YOLO("yolo26n.pt")

# Train using NDJSON dataset
results = model.train(data=path, epochs=100, imgsz=640)

Creating new Ultralytics Settings v0.0.6 file  
View Ultralytics Settings with 'yolo settings' or at 'C:\Users\sys_a\AppData\Roaming\Ultralytics\settings.json'
Update Settings with 'yolo settings key=value', i.e. 'yolo settings runs_dir=path/to/dir'. For help see https://docs.ultralytics.com/quickstart/#ultralytics-settings.


Matplotlib is building the font cache; this may take a moment.


Ultralytics 8.4.92  Python-3.13.2 torch-2.13.0+cu132 CUDA:0 (NVIDIA GeForce RTX 4050 Laptop GPU, 6140MiB)
engine\trainer: agnostic_nms=False, amp=True, angle=1.0, augment=False, auto_augment=randaugment, batch=16, bgr=0.0, box=7.5, cache=False, cfg=None, classes=None, close_mosaic=10, cls=0.5, cls_pw=0.0, cls_remap=True, compile=False, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=..\data\raw\plate.ndjson, degrees=0.0, deterministic=True, device=, dfl=1.5, dis=6.0, distill_model=None, dnn=False, dropout=0.0, dynamic=False, embed=None, end2end=None, epochs=100, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=640, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=yolo26n.pt, momentum=0.937, mosaic=1.0, multi_scale=0.0, name=train, nbs=64, nms=False, opset=None, optimize=False, optimizer=au

RuntimeError: Dataset '../data/raw/plate.ndjson' error  asyncio.run() cannot be called from a running event loop